### PR TITLE
[WIP] Add some missing blocks to block registry

### PR DESCRIPTION
**Pull request by Meronat**
Going through and adding some missing blocks to block registry, still have a lot more to add. 

Would like to attempt adding some new behaviors as well such as a `WalkedOnBehavior` or something for blocks like redstone ore, magma block, (pressure plates?), and similar. I'm not really sure if that is the correct decision though and if it is I don't know how to start going about it yet either. So if you have any input there, that would be highly appreciated.

### CODE REVIEW

**Comment by Cybermaxke** [comment]
You will also need to register all the block types in the `https://github.com/LanternPowered/LanternServer/blob/master/src/main/java/org/lanternpowered/server/data/io/store/item/ItemStackStore.java`. This is required to support the data values in `ItemStack`s, the data value will not always be the same as the block state id.
As for the `WalkedOnBehavior`, this could work for the redstone ores and magma blocks. But a different behavior `CollideWithBehavior` would probably be used for pressure plates, cactus, etc.

File: src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java [hunk 1]
```diff
@@ -862,16 +866,87 @@ public void registerDefaults() {
...
+                                .add(blastResistance(3.0)))
+                        .translation("tile.clay.name")
+                        .build("minecraft", "clay"));
         ////////////////////
         ///    Pumpkin   ///
         ////////////////////
         register(86, pumpkinBuilder()
                         .itemType(builder -> builder
-                                .properties(properties -> properties
+                            .properties(properties -> properties
```
Comment (Cybermaxke) [inline src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java]: There is some weird tabbing within the changes/additions you made, I will just comment a few with `+4 spaces` or `-4 spaces` where needed.

File: src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java [hunk 2]
```diff
@@ -890,6 +965,59 @@ public void registerDefaults() {
...
+                        .trait(LanternEnumTraits.DISGUISED_BLOCK_TYPE)
+                        .defaultState(state -> state
+                                .withTrait(LanternEnumTraits.DISGUISED_BLOCK_TYPE, LanternDisguisedBlockType.STONE).get())
+                        .itemType(builder -> builder
+                                .keysProvider(valueContainer -> valueContainer
+                                    .registerKey(Keys.DISGUISED_BLOCK_TYPE, LanternDisguisedBlockType.STONE)
+                                )
+                        )
+                        .properties(builder -> builder
+                            .add(hardness(0.75))
```
Comment (Cybermaxke) [inline src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java]: +4 spaces

File: src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java [hunk 3]
```diff
@@ -890,6 +965,59 @@ public void registerDefaults() {
...
+                        .build("minecraft", "monster_egg"),
+                blockState -> (byte) blockState.getTraitValue(LanternEnumTraits.DISGUISED_BLOCK_TYPE).get().getInternalId());
+        ////////////////////////
+        ///   Stone Bricks   ///
+        ////////////////////////
+        register(98, simpleBuilder()
+                        .trait(LanternEnumTraits.BRICK_TYPE)
+                        .defaultState(state -> state.withTrait(LanternEnumTraits.BRICK_TYPE, LanternBrickType.DEFAULT).get())
+                        .itemType(builder -> builder
+                            .keysProvider(valueContainer -> valueContainer.registerKey(Keys.BRICK_TYPE, LanternBrickType.DEFAULT)))
```
Comment (Cybermaxke) [inline src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java]: +4 spaces

File: src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java [hunk 4]
```diff
@@ -890,6 +965,59 @@ public void registerDefaults() {
...
+                                .add(hardness(1.5))
+                                .add(blastResistance(30)))
+                        .translation(TranslationProvider.of(LanternEnumTraits.BRICK_TYPE))
+                        .build("minecraft", "stonebrick"),
+                blockState -> (byte) blockState.getTraitValue(LanternEnumTraits.BRICK_TYPE).get().getInternalId());
+        ////////////////////////////////
+        ///   Brown Mushroom Block   ///
+        ////////////////////////////////
+        register(99, mushroomBlockBuilder()
+                .build("minecraft", "brown_mushroom_block"),
```
Comment (Cybermaxke) [inline src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java]: +8 spaces

File: src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java [hunk 5]
```diff
@@ -890,6 +965,59 @@ public void registerDefaults() {
...
+        ///   Red Mushroom Block   ///
+        //////////////////////////////
+        register(100, mushroomBlockBuilder()
+                        .build("minecraft", "red_mushroom_block"),
+            this::mushroomData);
+        ////////////////////
+        ///   End Stone  ///
+        ////////////////////
+        register(121, simpleBuilder()
+            .itemType()
```
Comment (Cybermaxke) [inline src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java]: +8 spaces for whole this section

File: src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java [hunk 6]
```diff
@@ -1090,6 +1238,71 @@ public void registerDefaults() {
...
                         .translation("tile.barrier.name")
                         .build("minecraft", "barrier"));
+        //////////////////////
+        ///   Prismarine   ///
+        //////////////////////
+        register(168, simpleBuilder()
+                        .trait(LanternEnumTraits.PRISMARINE_TYPE)
+                        .defaultState(state -> state.withTrait(LanternEnumTraits.PRISMARINE_TYPE, LanternPrismarineType.ROUGH).get())
+                        .itemType(builder -> builder.keysProvider(valueContainer -> valueContainer
+                            .registerKey(Keys.PRISMARINE_TYPE, LanternPrismarineType.ROUGH)))
```
Comment (Cybermaxke) [inline src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java]: `.keysProvider` starting on the next line to be consistent with the other registrations

File: src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java [hunk 7]
```diff
@@ -1305,6 +1568,22 @@ private byte pumpkinData(BlockState blockState) {
...
     }
 
+    private BlockTypeBuilder mushroomBlockBuilder() {
+        return simpleBuilder()
+                .itemType()
+                .trait(LanternEnumTraits.BIG_MUSHROOM_TYPE)
+                .defaultState(state -> state
+                        .withTrait(LanternEnumTraits.BIG_MUSHROOM_TYPE, LanternBigMushroomType.ALL_OUTSIDE).get())
+                .properties(builder -> builder
+                    .add(hardness(0.2))
```
Comment (Cybermaxke) [inline src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java]: +4 spaces

**Comment by Meronat** [comment]
Okay thanks for the comments. I'll add some more blocks and add them to the ItemStackStore hopefully later tonight. As for the spacing I'll fix those too and play with my formatter so it chooses the desired spacing more accurately.

**Pull request closed without merging**